### ISSUE TITLE
v3.1.0 memory leak when metricsInterval is 0

### ISSUE THREAD

**Issue by evgeni-k** [elastic/apm-agent-nodejs#1515]
**Describe the bug**
Memory leak detected when `metricsInterval` is set 0. You just start application and wait and then take a heap dump.

Repository: https://github.com/evgeni-k/apm-node-metrics-mem-leak

Config:
```
const APM = require(`elastic-apm-node`);
APM.start({
    metricsInterval: 0,
    metricsLimit: 100,
    serviceName: `mem-leak`,
    serverUrl: `http://localhost:8200`,
});
```

Production memory consumption graph:
![image](https://user-images.githubusercontent.com/29777598/68761070-57367f00-0624-11ea-9dfd-8885a25e8333.png)

Local heapdumps:
![image](https://user-images.githubusercontent.com/29777598/68761154-88af4a80-0624-11ea-91fc-3828abd9e788.png)

**To Reproduce**

1. start agent with `metricsInterval` 0.
2. wait
3. take a heap dump and analyze `Metrics` usage

**Expected behavior**

If set up to not send metrics then it will be great to not collect them at all.

**Environment (please complete the following information)**

- OS: Win10 x64
- Node.js 10.15.0
- APM Server: 7.4
- Agent vresion: 3.1.0

**How are you starting the agent? (please tick one of the boxes)**

- [x] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [ ] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Comment by evgeni-k** [comment]
if reporting is enabled then following code will not be executed https://github.com/elastic/apm-agent-nodejs/blob/v3.1.0/lib/metrics/reporter.js#L55. As a result loop monitor cannot be cleaned up https://github.com/elastic/apm-agent-nodejs/blob/v3.1.0/lib/metrics/runtime.js#L44

**Comment by lreuven** [comment]
Thanks for the catch!
Feel free to fix ( if not I'll get to it shortly).
Lior

**Comment by evgeni-k** [comment]
@lreuven could you please fix it and publish new package version?

**Comment by axw** [comment]
I've merged a fix - new release coming soon!

**Comment by axw** [comment]
The fix has been released in v3.2.0.
